Pegasus/sources/Integration.cpp

```diff
@@ -68,26 +70,82 @@ void IntegrateBody(
         double duration
     )
 {
-    glm::dvec3 const resultingAcceleration = ::IntegrateAcceleration(linearMotion.acceleration, linearMotion.force, material.GetInverseMass());
-    linearMotion.position = ::IntegratePosition(linearMotion.position, linearMotion.velocity, duration);
-    linearMotion.velocity = ::IntegrateVelocity(linearMotion.velocity, resultingAcceleration, duration);
-    linearMotion.velocity = ::IntegrateDamping(linearMotion.velocity, material.damping, duration);
+    glm::dvec3 const resultingAcceleration = ::IntegrateLinearAcceleration(
+        linearMotion.acceleration, linearMotion.force, material.GetInverseMass());
+    linearMotion.position = ::IntegrateLinearPosition(
+        linearMotion.position, linearMotion.velocity, duration);
+    linearMotion.velocity = ::IntegrateLinearVelocity(
+        linearMotion.velocity, resultingAcceleration, duration);
+    linearMotion.velocity = ::IntegrateLinearDamping(
+        linearMotion.velocity, material.damping, duration);
     linearMotion.force = glm::dvec3(0);
+
+    assert(!std::isnan(linearMotion.position.x + linearMotion.position.y + linearMotion.position.z));
+    assert(!std::isnan(linearMotion.velocity.x + linearMotion.velocity.y + linearMotion.velocity.z));
+    assert(!std::isnan(linearMotion.acceleration.x + linearMotion.acceleration.y + linearMotion.acceleration.z));
+    assert(!std::isnan(linearMotion.force.x + linearMotion.force.y + linearMotion.force.z));
```

Comment:
As of now `linearMotion.force` is always `glm::dvec3(0)`, will it be changed so that this assert actually tests something?